tests/stub/driver_parameters/test_connection_acquisition_timeout_ms.py

```diff
@@ -0,0 +1,211 @@
+    """
+    Connection Acquition Timeout Tests.
+
+    The connection acquisition timeout must account for the
+    whole acquisition execution time, whether a new connection is created,
+    an idle connection is picked up instead or we need to wait
+    until the full pool depletes.
+
+    In particular, the connection acquisition timeout (CAT) has precedence
+    over the socket connection timeout (SCT).
+
+    If the SCT is set to 2 hours and CAT to 50ms,
+    the connection acquisition should time out after 50ms,
+    even if the connection is successfully created within the SCT period.
+    """
+
+    required_features = (
+        types.Feature.BOLT_4_4,
+        types.Feature.CONNECTION_ACQUISITION_TIMEOUT
+    )
+
+    def setUp(self):
+        super().setUp()
+        self._server = StubServer(9001)
+        self._driver = None
+        self._session = None
+        self._sessions = []
+        self._txs = []
+
+    def tearDown(self) -> None:
+        self._server.reset()
+        for tx in self._txs:
+            with self.assertRaises(types.DriverError):
+                # The server does not accept ending the transaction.
+                # We still call it to potentially free resources.
+                tx.commit()
+
+        for s in self._sessions:
+            s.close()
+
+        if self._session:
+            self._session.close()
+
+        if self._driver:
+            self._driver.close()
+
+        return super().tearDown()
+
+    def test_should_works_when_every_step_is_done_in_time(self):
+        """
+        Everything in time scenario.
+
+        This test scenario tests the case where:
+
+        1. the connection acquisition timeout is higher than
+            the connection creation timeout
+        2. the connection is successfully created and in due time
+
+        Then the query is executed successfully
+        """
+        self._server.start(
+            self.script_path("session_run_auth_delay.script")
+        )
+
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+        uri = "bolt://%s" % self._server.address
+        self._driver = Driver(self._backend, uri, auth,
+                              connection_acquisition_timeout_ms=10000,
+                              connection_timeout_ms=5000)
+
+        self._session = self._driver.session("r")
+
+        list(self._session.run("RETURN 1 as n"))
+
+    def test_should_encompass_the_handshake_time(self):
+        """
+        Handshake tooks longer scenario.
+
+        This test scenario tests the case where:
+
+        1. the connection acquisition timeout is smaller than
+            the connection creation timeout
+        2. the connection is successfully created and in due time
+        3. the handshake tooks longer than the connection acquisition timeout
+
+        Then the query could not be executed since the connection acquisition
```

Comment:
```suggestion
        Then the query is not executed since the connection acquisition
```
(keeping everything in present tense, for consistency)